tools/GenerateCompilationDatabase.ps1

```diff
@@ -0,0 +1,166 @@
+Param(
+    [string]$Ms2ccVersion = "1.3.0",
+    [switch]$Update
+
+)
+
+$binlogFileBase = Join-Path (Split-Path $PSScriptRoot -parent) "BuildOutput\Binlogs"
+$binlogFiles = Get-ChildItem -Path $binlogFileBase -Filter *.binlog -Recurse | Select-Object -ExpandProperty FullName
+
+Write-Host "Binlog files found:"
+$binlogFiles | ForEach-Object { Write-Host $_ }
+
+$VCToolsInstallDir = . "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe" -Latest -prerelease -requires Microsoft.Component.MSBuild -property InstallationPath
+write-host "VCToolsInstallDir: $VCToolsInstallDir"
```

Comment:
Are you dumping variables or state/resources?

The latter is generally preferred. Doesn't matter to uses of the script if you name it `$msBuildPath`, `$msbuildexe`, `$msbuild_exe`, `$exe` or `$Fred`. But it can be relevant to the user the EXE you're going to execute

Reply:
This was the same pattern of `BuildAll.ps1`. I am writing the same information. I will change it to verbose only.